R/select.R

```diff
@@ -0,0 +1,155 @@
+#'  the value of the object in the expression (to be portable
+#'  between sessions). See the examples.
+#' @keywords datagen
+#' @concept preprocessing
+#' @concept variable_filters
+#' @export
+#' @examples
+#' rec <-
+#'   recipe(~., data = iris) %>%
+#'   step_select(Species, starts_with("Sepal"))
+#'
+#' prepped <- prep(rec, training = iris %>% slice(1:75))
+#'
+#' library(dplyr)
+#'
+#' dplyr_train <-
+#'   iris %>%
+#'   as_tibble() %>%
+#'   slice(1:75) %>%
+#'   select(Species, starts_with("Sepal"))
+#'
+#' rec_train <- bake(prepped, new_data = NULL)
+#' all.equal(dplyr_train, rec_train)
+#'
+#' dplyr_test <-
+#'   iris %>%
+#'   as_tibble() %>%
+#'   slice(76:150) %>%
+#'   select(Species, starts_with("Sepal"))
+#' rec_test <- bake(prepped, iris %>% slice(76:150))
+#' all.equal(dplyr_test, rec_test)
+#'
+#' # Embedding objects:
+#' sepal_vars <- c("Sepal.Width", "Sepal.Length")
+#'
+#' qq_rec <-
+#'   recipe(~., data = iris) %>%
+#'   # bad approach
+#'   step_select(Species, sepal_vars) %>%
+#'   # best approach
+#'   step_select(Species, !!sepal_vars) %>%
+#'   prep(training = iris)
+#'
+#' bake(qq_rec, new_data = NULL) %>% slice(1:4)
+#'
+#' # The difference:
+#' tidy(qq_rec, number = 1)
+#' tidy(qq_rec, number = 2)
+step_select <- function(recipe,
+                        ...,
+                        role = NA,
+                        trained = FALSE,
+                        inputs = NULL,
+                        skip = FALSE,
+                        id = rand_id("select")) {
+  add_step(
+    recipe,
+    step_select_new(
+      terms = ellipse_check(...),
+      trained = trained,
+      role = role,
+      inputs = inputs,
+      skip = skip,
+      id = id
+    )
+  )
+}
+step_select_new <- function(terms, role, trained, inputs, skip, id) {
+    step(
+      subclass = "select",
+      terms = terms,
+      role = role,
+      trained = trained,
+      inputs = inputs,
+      skip = skip,
+      id = id
+    )
+  }
+
+#' @export
+prep.step_select <- function(x, training, info = NULL, ...) {
+  sel <- eval_select_recipes(x$terms, training, info, allow_rename = TRUE)
+
+  step_select_new(
+    terms = x$terms,
+    trained = TRUE,
+    role = x$role,
+    inputs = sel,
+    skip = x$skip,
+    id = x$id
+  )
+}
+
+#' @export
+bake.step_select <- function(object, new_data, ...) {
+  dplyr::select(new_data, object$inputs)
+}
+
+
+print.step_select <-
+  function(x, width = max(20, options()$width - 35), ...) {
+    if (x$trained) {
+      cat(
+        "Variables selected ",
+        paste0(names(x$inputs), collapse = ", ")
+      )
+    } else {
+      cat(
+        "Terms selected ",
+        paste0(x$terms, collapse = ", ")
+      )
+    }
+    if (x$trained) {
+      cat(" [trained]\n")
+    } else {
+      cat("\n")
+    }
+    invisible(x)
+  }
+
+#' @rdname step_select
+#' @param x A `step_select` object
+#' @export
+tidy.step_select <- function(x, ...) {
+  if (is_trained(x)) {
+    res <- tibble(terms = x$inputs)
```

Comment:
This would change to `$terms`, but you might also consider unnaming it? Or using the `names()` here instead, if you want to show the renamed results instead of what was selected - up to you

Reply:
I went with the (new) names :+1: